data/transactions/logic/opcodes.go

```diff
@@ -61,38 +63,112 @@ const createdResourcesVersion = 6
 // experimental-
 const fidoVersion = 7 // base64, json, secp256r1
 
+type linearCost struct {
+	baseCost  int
+	chunkCost int
+	chunkSize int
+}
+
+func (lc *linearCost) compute(stack []stackValue) int {
+	cost := lc.baseCost
+	if lc.chunkCost != 0 && lc.chunkSize != 0 {
+		cost += lc.chunkCost * (len(stack[len(stack)-1].Bytes) + lc.chunkSize - 1) / lc.chunkSize
```

Comment:
@jannotti A couple of comprehension questions relating to: 
 `stackValue.Bytes` access without confirming `Bytes` is non-null.
* I see the pattern exists elsewhere, so the PR is _not_ introducing a new convention.
* What's the guiding rationale for when to check before accessing vs presuming an underlying value type?  
* Taking a step back:  _If_ a panic happens during AVM execution, how does algod behave?
  * I went up several layers and didn't spot an attempt to recover flow.  Possibly, I missed it.  In any event, it feels critical to understand the behavior.

Reply:
Panics are recovered
```
func eval(program []byte, cx *EvalContext) (pass bool, err error) {
	defer func() {
		if x := recover(); x != nil {
			buf := make([]byte, 16*1024)
			stlen := runtime.Stack(buf, false)
			pass = false
			errstr := string(buf[:stlen])
			if cx.Trace != nil {
				errstr += cx.Trace.String()
			}
			err = PanicError{x, errstr}
			cx.EvalParams.log().Errorf("recovered panic in Eval: %w", err)
		}
	}()
...
```
But we try not to panic.
The reason we don't need to check stack types here or in opcodes is that step() checks types first:
```
	// check args for stack underflow and types
	if len(cx.stack) < len(spec.Args) {
		return fmt.Errorf("stack underflow in %s", spec.Name)
	}
	first := len(cx.stack) - len(spec.Args)
	for i, argType := range spec.Args {
		if !opCompat(argType, cx.stack[first+i].argType()) {
			return fmt.Errorf("%s arg %d wanted %s but got %s", spec.Name, i, argType, cx.stack[first+i].typeName())
		}
	}
```

Reply:
PS We could probably get a speed boost by generating exactly the needed checks for each opcode and getting those checks inlined into the opcodes instead.  Same with the sanity checks after an opcode runs.

Reply:
@jannotti Thanks for these pointers + context - all makes sense.  Sorry, I traversed `eval` and missed the `recover` invocation.  Resolving.